content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -0,0 +1,456 @@
+ 
+Here is a Code snippet to illustrate the ```extends``` keyword:
+ 
+```java
+ 
+public class MainDog extends Dog {
+ 
+ //abstract method body definition
+ 
+ public void run(){
+ 
+     //in this class the method is implemented
+ 
+     System.out.println(“Dog is running”);
+ 
+ }
+ 
+ public static void main(Strings [] args){
+ 
+     MainDog dog = new MainDog();
+ 
+     //abstract method invocation
+ 
+     dog.run();
+ 
+ }
+ 
+}
+ 
+```
+ 
+#### Interface implementation
+An interface contains public, abstract method signatures (methods without implementation). It is also important to note that interfaces have no constructors and may contain final and static variables. Implementating abstraction by the interface method is possible by the use of the `implements` keyword.
+The child class inherits the parent class, overrides its method signatures, and provides a method body.
+ 
+```java
+ 
+public interface Car{
+ 
+ public void start();
+ 
+ public void stop();
+ 
+}
+ 
+public Benz implements Car{
+ 
+ /*@Override to override the stop method in the car class that has no implementation*/
+ 
+ @Override
+ 
+ public void start(){
+ 
+     System.out.println(“I started my Benz car ”);
+ 
+ }
+ 
+ /*to override the stop method in the car class that has no implementation*/
+ 
+ @Override
+ 
+ public void stop(){
+ 
+     System.out.println(“I stopped my Benz car”)
+ 
+ }
+ 
+}
+ 
+```
+ 
+To run this program, we need another class and the main method. The latter is the entrance to every Java application.
+ 
+```java
+ 
+public class Main{
+ 
+ public static void main(){
+ 
+     Car car = new Benz();
+ 
+     car.start();
+ 
+     car.stop();
+ 
+ }
+ 
+}
+ 
+```
+ 
+In the main method, `Car` represents the interface class, and `Benz` represents the concrete class or child class. The child class gives its memory to the `Car` object to access its implemented methods. The `Benz` class implemented the start and stop behaviors of the `Car` class in the manner it wanted. This makes our application robust. We can change or add requirements to our application without breaking down the application.
+ 
+For instance, we want to add another car that stops or starts differently. We can create a car, e.g. `Toyota`, that implements class `Car`, overrides its `stop()` and `start()` methods, implements them the way we want our `Toyota` to start and stop. There is no need to go into the `Car` class to change our code whenever there are new requirements.
+ 
+Now, that we’ve finally gotten abstraction off the list, on to the next one, encapsulation.
+ 
+### Encapsulation
+ 
+Imagine you are building an application, and the user can directly access and manipulate your data fields. How costly would that be? The outcome is better left to imagine than experience it.
+ 
+Apart from inheritance and polymorphism which I will write about in my next article. Encapsulation is another fundamental concept of OOP that every developer needs to get familiar with.
```

Comment:
> Apart from inheritance and polymorphism which I will write about in my next article. Encapsulation is another fundamental concept of OOP that every developer needs to get familiar with.

It doesn't make sense to have these two sentences separate when they are a part of the same idea/phrase. Revise this. I would put change the period between them to a comma like so:

> Apart from inheritance and polymorphism which I will write about in my next article, encapsulation is another fundamental concept of OOP that every developer needs to get familiar with.

I know the Hemingway tool will say that that sentence is way too long but personally I think it's fine. It makes more sense to have it as one sentence.

Reply:
Done